File: 1894-find-the-student-that-will-replace-the-chalk/1894-find-the-student-that-will-replace-the-chalk.rb
# @param {Integer[]} chalk
# @param {Integer} k
# @return {Integer}
def chalk_replacer(chalk, k)
  k %= chalk.sum # Remaining chalk after k full rounds

  chalk.each_index do |i|
    return i if k < chalk[i]
    k-=chalk[i]
  end
end

File: 1894-find-the-student-that-will-replace-the-chalk/1894-find-the-student-that-will-replace-the-chalk.js
/**
 * @param {number[]} chalk
 * @param {number} k
 * @return {number}
 */
var chalkReplacer = function(chalk, k) {
    k %= chalk.reduce((acc, cur) => acc + cur, 0) // chalk.sum
    for (let i=0;i<chalk.length;i++) {
        if (k < chalk[i]) return i;
        k -= chalk[i];
    }
};